src/serializer/ResidualFunctions.js

```diff
@@ -117,6 +117,51 @@ export class ResidualFunctions {
     if (!this.firstFunctionUsages.has(val)) this.firstFunctionUsages.set(val, bodyReference);
   }
 
+  _shouldUseFactoryFunction(funcBody: BabelNodeBlockStatement, instances: Array<FunctionInstance>) {
+    function shouldInlineFunction(): boolean {
+      let shouldInline = !funcBody;
+      if (!shouldInline && funcBody.start && funcBody.end) {
+        let bodySize = funcBody.end - funcBody.start;
+        shouldInline = bodySize <= 30;
+      }
+      return shouldInline;
+    }
+    let functionInfo = this.residualFunctionInfos.get(funcBody);
+    invariant(functionInfo);
+    let { usesArguments } = functionInfo;
+    return !shouldInlineFunction() && instances.length > 1 && !usesArguments;
+  }
+
+  _hasRewrritenFunctionInstance(
+    rewrittenAdditionalFunctions: Map<FunctionValue, Array<BabelNodeStatement>>,
+    instances: Array<FunctionInstance>
+  ): boolean {
+    return instances.find(instance => rewrittenAdditionalFunctions.has(instance.functionValue)) !== undefined;
```

Comment:
How does this scale? I'm always suspicious of linear lookups. Could instances grow with the size of the program we are prepacking?